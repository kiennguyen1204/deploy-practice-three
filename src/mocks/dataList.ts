export const dataItemDetail = {
  effects: [
    {
      title: 'effects',
      image: '/src/assets/images/products/color_filter.png',
      content: 'Calming, Creative, Happy, Relaxing, Sleepy, Uplifting'
    },
    {
      title: 'relieve',
      image: '/src/assets/images/products/brain.png',
      content:
        'Anxiety, Arthritis, Chronic Pain, Depression, Fatigue, Inflammation, Insomnia, Irregular Bowel Movements, Migraines, Mood Swings'
    },
    {
      title: 'aromas',
      image: '/src/assets/images/products/carbon_smoke.svg',
      content: 'Chemical, Citrus, Earthy, Pungent, Sour'
    }
  ]
};

export const DATA_LIST = [
  {
    isSoldOut: 86,
    title: '',
    totalReviews: 135,
    discount: '$200.00',
    price: '$102.00',
    unit: 'unit 1',
    volume: [],
    image: 'https://tinyurl.com/2s34y377',
    sale: 'Indica 70%',
    description:
      'Jungle Diamonds is a slightly indica dominant hybrid strain (60% indica/40% sativa) created through crossing the infamous Slurricane X Gorilla Glue #4 strains.',
    category: 'CONCENTRATES',
    name: 'Mix And Match Shatter/Budder 28g (4 X 7G)',
    rate: '46/5',
    id: '1'
  },
  {
    isSoldOut: 86,
    title: '',
    totalReviews: 135,
    discount: '$200.00',
    price: '$102.00',
    unit: 'unit 1',
    volume: [],
    image: 'https://tinyurl.com/2s34y377',
    sale: 'Indica 70%',
    description:
      'Jungle Diamonds is a slightly indica dominant hybrid strain (60% indica/40% sativa) created through crossing the infamous Slurricane X Gorilla Glue #4 strains.',
    category: 'CONCENTRATES',
    name: 'Mix And Match Shatter/Budder 28g (4 X 7G)',
    rate: '46/5',
    id: '1'
  },
  {
    isSoldOut: 86,
    title: '',
    totalReviews: 135,
    discount: '$200.00',
    price: '$102.00',
    unit: 'unit 1',
    volume: [],
    image: 'https://tinyurl.com/2s34y377',
    sale: 'Indica 70%',
    description:
      'Jungle Diamonds is a slightly indica dominant hybrid strain (60% indica/40% sativa) created through crossing the infamous Slurricane X Gorilla Glue #4 strains.',
    category: 'CONCENTRATES',
    name: 'Mix And Match Shatter/Budder 28g (4 X 7G)',
    rate: '46/5',
    id: '1'
  },
  {
    isSoldOut: 86,
    title: '',
    totalReviews: 135,
    discount: '$200.00',
    price: '$102.00',
    unit: 'unit 1',
    volume: [],
    image: 'https://tinyurl.com/2s34y377',
    sale: 'Indica 70%',
    description:
      'Jungle Diamonds is a slightly indica dominant hybrid strain (60% indica/40% sativa) created through crossing the infamous Slurricane X Gorilla Glue #4 strains.',
    category: 'CONCENTRATES',
    name: 'Mix And Match Shatter/Budder 28g (4 X 7G)',
    rate: '46/5',
    id: '1'
  }
];

export const LIST_PRODUCT_CATEGORY_FILTER = [
  {
    name: 'Sales',
    value: 12
  },
  {
    name: 'Cannabis',
    value: 430
  },
  {
    name: 'Pre-Rolls',
    value: 40
  },
  {
    name: 'CBD Oil',
    value: 12
  },
  {
    name: 'Magic Mushrooms',
    value: 34
  },
  {
    name: 'Extracts',
    value: 26
  },
  {
    name: 'Edibles',
    value: 32
  },
  {
    name: 'Vape Pens',
    value: 12
  },
  {
    name: 'Accessories',
    value: 10
  },
  {
    name: 'Bath & Body',
    value: 8
  },
  {
    name: 'Bundles',
    value: 24
  },
  {
    name: 'Wholesale',
    value: 28
  }
];
