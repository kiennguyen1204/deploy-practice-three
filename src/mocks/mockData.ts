import { Product } from 'interfaces/item';

export const DATA_ITEM = {
  image: 'test-image.jpg',
  subtitle: 'Test Subtitle',
  title: 'Test Title',
  description: 'Test Description',
  to: '/test-link'
};

export const MOCK_PROPS = {
  image: '/sample_image.png',
  title: 'Sample Title',
  description: 'Sample Description',
  to: '/sample-link',
  shop: 'Sample Shop',
  icon: '/sample_icon.png'
};

export const MOCK_ITEM: Product = {
  id: '1',
  name: 'Product A',
  title: 'Product A Title',
  description: 'Description for Product A',
  sale: '30%',
  image: 'image-url-1',
  reviews: 15,
  isSoldOut: false,
  category: 'Category A',
  price: 100,
  discount: 10,
  volume: ['100ml', '200ml'],
  totalReviews: 20,
  quantity: 50
};

export const MOCK_DATA_CUSTOMER = {
  userName: 'Vikki Starr',
  avatar: 'assets/images/reviews/VikkiStarr.png',
  comment: `Absolutely love TopShelfBC; affordable on any budget and such fast delivery, straight to my door! I recommend them to all my friends and family for their 420 needs.`,
  star: 4,
  date: 'January 15, 2023'
};

export const DATA_CART = {
  total: 305.0,
  shipping: 50,
  payTotal: 355,
  discount: 0,
  cart: [
    {
      name: 'Mix And Match Shatter/Budder 28g (4 X 7G)',
      image: 'assets/images/products/snack_yellow.webp',
      price: 120.0,
      amount: 2,
      total: 240.0,
      integra: [
        {
          namePack: 'Add Integra Pack - 4g',
          amount: 1,
          price: 2.0,
          total: 2.0
        },
        {
          namePack: 'Add Integra Pack - 8g',
          amount: 1,
          price: 3.0,
          total: 3.0
        }
      ],
      subTotal: 245.0
    },
    {
      name: 'Mix And Match Shatter/Budder 28g (4 X 7G)',
      image: 'assets/images/products/snack_purple.webp',
      price: 13.0,
      amount: 5,
      total: 65.0,
      integra: [],
      subTotal: 65.0
    }
  ]
};
