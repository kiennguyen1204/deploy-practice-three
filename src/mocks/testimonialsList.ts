export const TESTIMONIALS_LIST = [
  {
    userName: 'Vikki Starr',
    avatar: '/src/assets/images/reviews/VikkiStarr.png',
    comment: `Absolutely love TopShelfBC; affordable on any budget and such fast delivery, straight to my door! I recommend them to all my friends and family for their 420 needs.`,
    star: 4,
    date: 'January 15, 2023'
  },
  {
    userName: 'Vikki Starr',
    avatar: '/src/assets/images/reviews/VikkiStarr.png',
    comment: `Absolutely love TopShelfBC; affordable on any budget and such fast delivery, straight to my door! I recommend them to all my friends and family for their 420 needs.`,
    star: 4,
    date: 'January 15, 2023'
  },
  {
    userName: 'Vikki Starr',
    avatar: '/src/assets/images/reviews/VikkiStarr.png',
    comment: `Absolutely love TopShelfBC; affordable on any budget and such fast delivery, straight to my door! I recommend them to all my friends and family for their 420 needs.`,
    star: 5,
    date: 'January 15, 2023'
  },
  {
    userName: 'Vikki Starr',
    avatar: '/src/assets/images/reviews/VikkiStarr.png',
    comment: `Absolutely love TopShelfBC; affordable on any budget and such fast delivery, straight to my door! I recommend them to all my friends and family for their 420 needs.`,
    star: 5,
    date: 'January 15, 2023'
  },
  {
    userName: 'Vikki Starr',
    avatar: '/src/assets/images/reviews/VikkiStarr.png',
    comment: `Absolutely love TopShelfBC; affordable on any budget and such fast delivery, straight to my door! I recommend them to all my friends and family for their 420 needs.`,
    star: 5,
    date: 'January 15, 2023'
  },
  {
    userName: 'Vikki Starr',
    avatar: '/src/assets/images/reviews/VikkiStarr.png',
    comment: `Absolutely love TopShelfBC; affordable on any budget and such fast delivery, straight to my door! I recommend them to all my friends and family for their 420 needs.`,
    star: 5,
    date: 'January 15, 2023'
  }
];
