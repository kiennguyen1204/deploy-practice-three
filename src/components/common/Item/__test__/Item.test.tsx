import { render, screen, fireEvent } from '@testing-library/react';

// mocks
import { MOCK_ITEM } from 'mocks/mockData';

// components
import ItemCard, { Props } from 'components/common/Item';

jest.mock('react-router-dom', () => ({
  Link: ({ to, children }: { to: string; children: React.ReactNode }) => <a href={to}>{children}</a>
}));

const setup = (props = {}) => {
  const defaultProps: Props = {
    item: MOCK_ITEM,
    index: 0,
    className: 'test-class',
    onClick: jest.fn(),
    ...props
  };

  render(<ItemCard {...defaultProps} />);
};

describe('ItemCard', () => {
  it('renders the item card with correct data', () => {
    setup();
    expect(screen.getByText(MOCK_ITEM.category)).toBeInTheDocument();
  });

  it('renders the "Sold Out" element when isSoldOut is true', () => {
    setup();

    const soldOutElement = screen.queryByTestId('sold-out');
    expect(soldOutElement).toBeInTheDocument();
  });

  it('does not render the "Sold Out" element when isSoldOut is false', () => {
    const mockItemNotSoldOut = { ...MOCK_ITEM, isSoldOut: false };
    setup({ item: mockItemNotSoldOut });

    const soldOutElement = screen.queryByTestId('sold-out');
    expect(soldOutElement).toBeNull();
  });

  it('calls the onClick function when "Add to Cart" button is clicked', () => {
    const onClickMock = jest.fn();
    render(<ItemCard item={MOCK_ITEM} index={0} className='test-class' onClick={onClickMock} />);

    const addToCartButton = screen.getByText('Add to Cart');
    fireEvent.click(addToCartButton);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it('ItemCard renders correctly', () => {
    const { container } = render(
      <ItemCard item={MOCK_ITEM} index={0} className='test-class' onClick={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });
});
