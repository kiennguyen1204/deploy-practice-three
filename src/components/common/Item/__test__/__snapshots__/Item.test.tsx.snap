// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard ItemCard renders correctly 1`] = `
<div>
  <article
    class="item-card test-class"
    data-testid="item-card-1"
  >
    <a
      href="/products/1"
    >
      <section
        class="image-card"
      >
        <img
          alt=""
          data-testid="item-image"
          src="image-url-1"
        />
      </section>
      <section
        class="content-item"
      >
        <p
          class="cate-item"
          data-testid="category"
        >
          Category A
        </p>
        <h2
          class="title-item"
          data-testid="name"
        >
          Product A
        </h2>
        <div
          class="star-review"
        >
          <span
            class="voted"
          >
            <img
              alt="Star"
              data-testid="star-image"
              src="IMAGE_MOCK"
            />
            /
            5
          </span>
          <span
            class="review"
            data-testid="reviews"
          >
            15
            <span
              class="reviewsText"
            >
              Reviews
            </span>
          </span>
        </div>
        <div
          class="sale"
          data-testid="sale"
        >
          30%
        </div>
        <div
          class="discount-n-price"
          data-testid="discount"
        >
          <p
            class="discount"
          >
            $
            10
          </p>
          <p
            class="price"
            data-testid="price"
          >
            100
          </p>
        </div>
        <div
          class="volumes"
          data-testid="volumes"
        >
          <div
            class="volume-child"
            data-testid="volume-0"
          >
            100ml
          </div>
          <div
            class="volume-child"
            data-testid="volume-1"
          >
            200ml
          </div>
        </div>
      </section>
    </a>
    <div
      class="flex-container button-buy"
    >
      <button
        class="btn btn-add-cart"
        data-testid="add-to-cart-button"
      >
        Add to Cart
      </button>
    </div>
  </article>
</div>
`;
